_pylibczi/_pylibczi.cpp

```diff
@@ -207,6 +229,104 @@ static PyObject *cziread_allsubblocks(PyObject *self, PyObject *args) {
     return Py_BuildValue("OO", images, (PyObject *) coordinates);
 }
 
+// Begin new code
+
+ static int convertDictToPlaneCoords(PyObject *obj, void * dim_p){
+    if(!PyDict_Check(obj)){ // docs says it returns true/false but it returns an integer
+        return 0; // not a dictionary somethings wrong
+    };
+
+    libCZI::CDimCoordinate *dims = static_cast<libCZI::CDimCoordinate *>(dim_p);
+    std::map < std::string, libCZI::DimensionIndex > tbl{
+            {"V", libCZI::DimensionIndex::V},
+            {"H", libCZI::DimensionIndex::H},
+            {"I", libCZI::DimensionIndex::I},
+            {"S", libCZI::DimensionIndex::S},
+            {"R", libCZI::DimensionIndex::R},
+            {"T", libCZI::DimensionIndex::T},
+            {"C", libCZI::DimensionIndex::C},
+            {"Z", libCZI::DimensionIndex::Z}
+    };
+    std::for_each(tbl.begin(), tbl.end(), [&](std::pair< const std::string, libCZI::DimensionIndex > &kv){
+        PyObject *pyLetter;
+        PyObject *pyInt;
+        pyLetter = Py_BuildValue("s",kv.first.c_str());
+        if(PyDict_Contains(obj, pyLetter)){
+            int dimValue = 0;
+            pyInt = PyDict_GetItem(obj, pyLetter);
+            dimValue = static_cast<int>(PyLong_AsLong(pyInt));
+            dims->Set(kv.second, dimValue);
+        }
+        Py_DecRef(pyLetter);
+    });
+    return 1; // success
+}
+
+static int listToIntRect(PyObject *obj, void * rect_p){
+    libCZI::IntRect *rect = static_cast<libCZI::IntRect *>(rect_p);
+    if( obj == Py_None ){
+        rect->x = rect->y = rect->w = rect->h = 0;
+        return 1;
+    }
+
+    if(!PyTuple_Check(obj)){
+        std::cout << "region not a Tuple!" << std::endl;
+    }
+
+    if(PyTuple_Size(obj) != 4){
+        std::cerr << "Error: format for region (x, y, w, h)! " << std::endl;
+        return 0;
+    }
+
+    rect->x = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 0)));
+    rect->y = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 1)));
+    rect->w = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 2)));
+    rect->h = static_cast<int>(PyLong_AsLong(PyTuple_GetItem(obj, 3)));
+    return 1;
+}
+
+static PyObject *cziread_mosaic(PyObject *self, PyObject *args) {
+    char *filename_buf;
+    float scaleFactor;
+    libCZI::CDimCoordinate planeCoord;
+    libCZI::IntRect imBox{0,0,0,0};
+    // parse arguments
+    if (!PyArg_ParseTuple(args, "sO&O&f", &filename_buf,
+                          &convertDictToPlaneCoords, &planeCoord,
+                          &listToIntRect, &imBox,
+                          &scaleFactor)
+            ){
+        std::cout << "conversion of sOOf failed!" << std::endl;
+        return nullptr;
+    }
+
+    auto cziReader = open_czireader_from_cfilename(filename_buf);
+    auto statistics = cziReader->GetStatistics();
+    // handle the case where the function was called with no selection rectangle
+    if ( imBox.w == 0 || imBox.h == 0 ) imBox = statistics.boundingBox;
```

Comment:
this condition would be stronger using && instead of ||

Reply:
Maybe I'm missing something but it seems like the opposite is true?
What does it mean to ask for an image with 0, height 1000  or width 1000, heigh 0?

Reply:
I was only concerned about `None` --> (0,0,0,0) --> full bounds.  If they specify a width of 0 or height of 0, maybe they should only get a warning instead of loading the whole image?